app/controllers/api/current/exotel_call_sessions_controller.rb

```diff
@@ -6,15 +6,37 @@ def create
       respond_in_plain_text(:bad_request) and return
     end
 
-    session = CallSession.new(params[:From], parse_patient_phone_number)
+    session = CallSession.new(params[:CallSid], params[:From], parse_patient_phone_number)
     if session.authorized?
-      session.save(params[:CallSid])
+      session.save
       respond_in_plain_text(:ok)
     else
       respond_in_plain_text(:forbidden)
     end
   end
 
+  def fetch
+    session = CallSession.fetch(params[:CallSid])
+
+    if session.present?
+      respond_in_plain_text(:ok, session.patient_phone_number.number)
+    else
+      respond_in_plain_text(:not_found)
+    end
+  end
+
+  def terminate
+    session = CallSession.fetch(params[:CallSid])
+
+    if session.present?
+      session.kill
+      report_call_info
```

Comment:
This can be an after_action hook

Reply:
We had this as an after_action, but ended up changing it because:
* only used by one method
* used only if `session.present?` is `true`